### PR TITLE
added method to remove gestures + missing delegate method

### PR DESCRIPTION
**Pull request by 2020Deception**
helps if you want to remove the gestures at a given point. Also added a delegate method that was causing a warning on stock compiler settings.

### CODE REVIEW

File: SWRevealViewController/SWRevealViewController.m [hunk 1]
```diff
@@ -896,6 +899,10 @@ - (UIPanGestureRecognizer*)panGestureRecognizer
     return _panGestureRecognizer;
 }
 
+- (void)removeGestureRecognizers {
+    _panGestureRecognizer = nil;
+    _tapGestureRecognizer = nil;
+}
```
Comment (cemaleker) [inline SWRevealViewController/SWRevealViewController.m:905]: At this point recognizers are still attached to the views and will trigger their actions if they recognize anything. What is the purpose of this method actually?

**Comment by iDevelopper** [comment]
What is it for?

You can implement the delegate methods:

// Implement this to return NO when you want the pan gesture recognizer to be ignored
- (BOOL)revealControllerPanGestureShouldBegin:(SWRevealViewController *)revealController;

// Implement this to return NO when you want the tap gesture recognizer to be ignored
- (BOOL)revealControllerTapGestureShouldBegin:(SWRevealViewController *)revealController;

**Comment by 2020Deception** [comment]
its not performant to do this especially they aren't needed, and there is no way to remove or override a gesture recognizer. I'm gong to close this as it seems you all don't get my drift.

**Comment by iDevelopper** [comment]
I'm not the owner of this repo. Then I can't merge anything!

**Comment by iDevelopper** [comment]
You can also set the gesture's property enabled to false when you don't need it. Or don't call them at all. Or remove them (removeGestureRecognizer) from the view.

**Pull request closed without merging**